apps/server/src/modules/board/board-context-api-helper.service.ts

```diff
@@ -0,0 +1,44 @@
+import { AuthorizationService } from '@modules/authorization';
+import { BoardExternalReference, BoardExternalReferenceType, BoardNodeService, ColumnBoard } from '@modules/board';
+import { CourseService } from '@modules/learnroom';
+import { RoomService } from '@modules/room';
+import { Injectable } from '@nestjs/common';
+import { User } from '@shared/domain/entity';
+import { EntityId } from '@shared/domain/types';
+
+@Injectable()
+export class BoardContextApiHelperService {
+	constructor(
+		private readonly authorizationService: AuthorizationService,
+		private readonly courseService: CourseService,
+		private readonly roomService: RoomService,
+		private readonly boardNodeService: BoardNodeService
+	) {}
+
+	public async getSchoolIdForBoardNode(userId: EntityId, nodeId: EntityId): Promise<EntityId> {
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
```

Comment:
this doesnt seem to fit into this function. user shouldnt have anything todo with the schoolid of a boardnode.

Reply:
refactored